### PR TITLE
feat(common): support ICU standard "quarters (format + standalone)" with `DatePipe`

### LINKED ISSUE #49175: I want to add `Q` for cldr
### Which @angular/* package(s) are relevant/related to the feature request?

_No response_

### Description

I saw that there was a TODO from two years ago

[`TODO(ocombe): support all missing cldr formats: U, Q, D, F, e, j, J, C, A, v, V, X, x `](https://github.com/angular/angular/blob/3d589d8bdeb81e953d1c3b6516cebcf07c214f52/packages/common/src/i18n/format_date.ts#L448)

I want to know if anyone is working on this implementation now. 

### Proposed solution

If no one is working on this I can implement for `Q`

### Alternatives considered

Someone else to implement :)

### PR DESCRIPTION
**Pull request by avmaxim**
This commit adds support for quarters `q/qq/qqq/qqqq/qqqqq` and `Q/QQ/QQQ/QQQQ/QQQQQ` date formats in the `DatePipe`, as well as extends all locales with format and standalone quarter names.

Fixes #49175

Apart from that this PR updates the following auto-generated files based on what `generate-locales-tool` on top of `CLDR` produces:
* `packages/core/src/i18n/locale_en.ts`
* `packages/common/locales/closure-locale.ts`
* `packages/examples/common/pipes/ts/locale-fr.ts`

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes

## Does this PR introduce a breaking change?

- [x] Yes (needs confirmation from Angular team)
- [ ] No

Since this PR extends all locales with format and standalone quarter names, the order of data fields from the locale data array has changed. See an example in `packages/examples/common/pipes/ts/locale-fr.ts`.

Taking it into account, the developers are required to make sure they properly use:
1. Locale data in all locale files under `@angular/common/locales/*` 
2. Enum `ɵLocaleDataIndex` located at `@angular/core` that represents the index of each type of locale data in locale data array

### CODE REVIEW

**Comment by avmaxim** [comment]
@petebacondarwin @devversion @AndrewKushnir @atscott 
Hi everyone! 👋🏻  I've implemented the feature to support quarters in `@angular/common` module. Please take a look and let me kindly know if we need anything else in terms of this PR. I'll be glad to cooperate and do my best to boost its merging! 🙂 Feel free to add anyone else here who can also help us on it. Thank you!

File: packages/common/src/pipes/date_pipe.ts [hunk 1]
```diff
@@ -125,6 +125,16 @@ export const DATE_PIPE_DEFAULT_OPTIONS =
  *  |                     | YY          | Numeric: 2 digits + zero padded                               | 02, 20, 01, 17, 73                                         |
  *  |                     | YYY         | Numeric: 3 digits + zero padded                               | 002, 020, 201, 2017, 20173                                 |
  *  |                     | YYYY        | Numeric: 4 digits or more + zero padded                       | 0002, 0020, 0201, 2017, 20173                              |
+ *  | Quarter             | Q           | Numeric: 1 digit                                              | 2                                                          |
+ *  |                     | QQ          | Numeric: 2 digits + zero padded                               | 02                                                        |
```
Comment (jerone) [inline packages/common/src/pipes/date_pipe.ts:129]: Nitpic:
image

**Comment by RuBAN-GT** [comment]
Ooo, very important PR. @avmaxim thank you for your initiative!

File: packages/common/src/pipes/date_pipe.ts (hunk 1 above)
Comment (avmaxim) [reply packages/common/src/pipes/date_pipe.ts:129]: @jerone Thanks for the review! Appriciate it! Yeah, will take care of that in a little while. Seems like those borders passed out of the viewport:

image

**Comment by avmaxim** [comment]
@AndrewKushnir @petebacondarwin @devversion Hi 👋🏻 Could you please take a look at it once you have time or add anyone who can review it too and pinpoint any bottlenecks to work on and update here? It would be really nice to have this feature in the master with any new major update.

**Comment by RuBAN-GT** [comment]
I'm wondering why this PR still open. Can somebody approve it?

**Comment by devversion** [comment]
@avmaxim Thanks for the PR. The contribution is much appreciated and actually looks great!

I've checked more with the team and we currently feel like we would prefer to not doing that for now, given us looking into potentially using the browser `Intl` API. Adding more ICU symbols here might make such a future migration a little more difficult, while there isn't a significant need for e.g. supporting quarter formatting (judging by the zero upvotes in the feature request)- while also seeing that there are many more symbols we don't support.

https://github.com/angular/angular/issues/49143
https://github.com/tc39/ecma402/issues/374

We can definitely discuss this further, but at this point, I think it makes sense to hold off. If you are willing, it would be great to remove the TODO asking for more ICU symbols to be supported. It's been such a long time and this is not a priority (and not necessarily what we want to do given the low user requests/future potential plans).

**Pull request closed without merging**